src/lib/support/Pool.h

```diff
@@ -0,0 +1,125 @@
+/*
```

Comment:
This should also contain the original copyright since it's heavily leveraged from that base of code.

Reply:
done